src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -0,0 +1,147 @@
+package nextstep.subway.line.domain;
+
+import nextstep.subway.station.domain.Station;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+@Embeddable
+public class Sections {
+    @OneToMany(mappedBy = "line", cascade = {CascadeType.PERSIST, CascadeType.MERGE}, orphanRemoval = true)
+    private List<Section> sections = new ArrayList<>();
+
+    public Sections() {
+    }
+
+    public Sections(Line line, Station upStation, Station downStation, int distance) {
+        Section section = new Section(line, upStation, downStation, distance);
+        this.sections.add(section);
+    }
+
+    public void addSection(Line line, Station upStation, Station downStation, int distance) {
+        List<Station> stations = getStations();
+
+        boolean isUpStationExisted = isStationExisted(upStation, stations);
+        boolean isDownStationExisted = isStationExisted(downStation, stations);
+
+        validate(stations, isUpStationExisted, isDownStationExisted);
+
+        if (stations.isEmpty()) {
+            sections.add(new Section(line, upStation, downStation, distance));
+            return;
+        }
+
+        if (isUpStationExisted) {
+            updateUpstation(upStation, downStation, distance);
+            sections.add(new Section(line, upStation, downStation, distance));
+
+        } else if (isDownStationExisted) {
+            updateDownStation(upStation, downStation, distance);
+            sections.add(new Section(line, upStation, downStation, distance));
+
+        } else {
+            throw new RuntimeException();
+        }
+    }
+
+    public List<Station> getStations() {
+        if (sections.isEmpty()) {
+            return Arrays.asList();
```

Comment:
빈 리스트는 아래의 싱글턴 객체를 활용하는게 어떨까요? 

```suggestion
            return Collections.emptyList();
```